primer/brand
# ActionMenu text reflow fix

## Summary

Resolves https://github.com/github/primer/issues/3758 (SEV-1 ⚠️ )

Removes truncation of long text in `ActionMenu` button and lists, allowing it to instead reflow [per WCAG guidelines](https://www.w3.org/TR/WCAG22/#reflow).

## List of notable changes:



- Removed fixed height, text truncation and ellipsis on buttons to allow text to flow onto multiple lines if needed
- Fixed issues with inherited paddings, which weren't previously accurate
- Removed truncation and ellipsis on popup list items
- Widths on action list items are now dynamically calculated


## Steps to test:



1. [Use this visual diff](https://github.com/primer/brand/compare/rezrah/action-menu-a11y-reflow?expand=1#diff-9aece6da8765c3bade01d153faea0fae7d95ea602525d171f3bf1baae0c35266) to see the before/after of this change. The story name will change after reviews are in, because the diff won't work if that's done first.

## Supporting resources (related issues, external links, etc):

- https://github.com/github/primer/issues/3758

## Contributor checklist:


## Reviewer checklist:


## Screenshots:

> Please try to provide before and after screenshots or videos

<table>
<tr>
<th> Before</th> <th> After </th>
</tr>
<tr>
<td valign="top">

![Screenshot 2024-08-16 at 17 35 16](https://github.com/user-attachments/assets/6b57ca3d-5e7a-4403-8c87-7750e7aa9739)


 </td>
<td valign="top">

![Screenshot 2024-08-16 at 17 35 00](https://github.com/user-attachments/assets/e744ba76-81cb-48e0-89a8-148348766211)


</td>
</tr>
</table>
